HW13/HW13.cpp

```diff
@@ -0,0 +1,242 @@
+
+    std::string currentDay = std::to_string(getDay(now)) + "-" + std::to_string(getMonth(now)) + "-" + std::to_string(getYear(now));
+    
+    return currentDay;
+}
+
+bool todayWasGame()
+{
+    std::string currentDay = getCurrentDay();
+
+    std::string dayFromFile;
+    std::ifstream outputfile(filename);
+    
+    while (std::getline(outputfile, dayFromFile)) { 
+        if (currentDay == dayFromFile)
+        {
+            outputfile.close();
+            return true;
+        }
+    }
+    outputfile.close();
+
+    return false;
+}
+
+void writeСurrentDate()
+{
+    std::string currentDay = getCurrentDay();
+
+    std::ofstream inputfile(filename);
+
+    if (inputfile.is_open()) {
+        inputfile << currentDay << std::endl;
+        inputfile.close();
+    }
+}
+
+std::string toUpper(std::string wordle)
+{
+    std::string result;
+    for (int i = 0; i < wordle.length(); i++)
+    {
+        result += std::toupper(wordle[i]);
+    }
+    return result;
+}
+
+std::string tolower(std::string wordle)
+{
+    std::string result;
+    for (int i = 0; i < wordle.length(); i++)
+    {
+        result += std::tolower(wordle[i]);
+    }
+    return result;
+}
+
+bool getResult(std::string& result, std::string wordle, std::string userWordle)
+{
+    wordle = tolower(wordle);
+    userWordle = tolower(userWordle);
+
+    if (wordle == userWordle)
+    {
+        result = toUpper(wordle);
+        return true;
+    }
+
+    result = "*****";
+
+    for (int i = wordle.length() - 1; i >= 0; i--)
+    {
+        if (wordle[i] == userWordle[i])
+        {
+            result[i] = std::toupper(wordle[i]);
+        }
+    }
+
+    for (int i = 0; i < wordle.length(); i++)
+    {
+        if (result[i] == '*')
+        {
+            std::string resLower = tolower(result);
+
+            int countWordle = std::count(wordle.begin(), wordle.end(), userWordle[i]);
+            int countResult = std::count(resLower.begin(), resLower.end(), userWordle[i]);
+
+            if (countWordle > 0 && countWordle - countResult > 0)
+            {
+                result[i] = userWordle[i];
+            }
+        }
+    }
+    return false;
+}
+
+void startGame()
+{
+    int userSelect = 2;
+    std::string wordle;
+    std::string userWordle;
+    std::string result = "*****";
+    
+    bool wasGame = todayWasGame();
+
+    while (userSelect != 0)
+    {
+        int numCount = 0;
+        std::cout << "1 - Random wordle" << std::endl;
+        if (!wasGame)
+        {
+            std::cout << "2 - Wordle of the day" << std::endl;
+        }
+        std::cout << "0 - Exit" << std::endl;
+        std::cout << "Enter: ";
+        std::cin >> userSelect;
+        std::cout << std::endl;
+
+        switch (userSelect)
+        {
+        case 1:
+            wordle = getRandomWordle();
+            break;
+        case 2:
+            if (wasGame)
+            { 
+                continue; 
+            }
+            wordle = getWordleFromFile();
```

Comment:
reading it from file into vector every time is quite inefficient. One time per launch would be more than enough reusing then already setup vector of words